apps/course-builder-web/src/inngest/functions/ocr/ocr-code-extractor.ts

```
@@ -12,10 +13,18 @@ export const perform_code_extraction = inngest.createFunction(
   },
   async ({ event, step }) => {
     const screenshotUrl = event.data.ocrWebhookEvent.screenshotUrl as string
-    console.log('\t>>>> screenshotUrl: ', screenshotUrl)
-    await step.run('extract code from screenshot', async () => {
-      const codestring = await extractCodeFromScreenshot(screenshotUrl).catch((e) => console.error(e))
-      return { codestring }
+
+    return await step.run('extract code from screenshot', async () => {
+      return extractCodeFromScreenshot(screenshotUrl)
+        .then((codeString) => {
+          return {
+            codeString,
+          }
+        })
+        .catch((e) => {
+          console.error(e)
+          throw new NonRetriableError('Error extracting code from screenshot')
```

instead of swallowing the error I'm rethrowing and making it "non-retriable" - actually probably want to just let it throw so you see the error in inngest at the step and it will retry as needed